### DISCUSSION TITLE
TLS management on the FTP server

### DISCUSSION THREAD

**Discussion by nassissme** [drakkan/sftpgo#1356] [General]
Hi, we're using sftpgo for FTP on port 21 and for SFTP on port 22. We needed to add FTPES management on port 990, which changed the behavior of port 21.

Before :
```
FTP server
Status: "Started"

Address: ":21"
TLS: "Disabled"
Passive IP: ******

Passive port range: "******-******"
```

After :
```
FTP server
Status: "Started"

Address: ":21"
TLS: "Plain and explicit"
Passive IP: ******

Address: ":990"
TLS: "Explicit required"
Passive IP: ******

Passive port range: "******-******"
```

We want to keep port 21 TLS Disabled. 

After analyzing the ftp server code, I can see that if one of the ports (bindings) contains a certificate, then all the ports are configured for TLS. 

Is this the desired behavior? Is it possible to modify this behavior in order to configure in TLS only the bindings with the `tls_mode` parameter different from 0? 

Thank you.

**Comment by drakkan** [comment]
https://github.com/drakkan/sftpgo/commit/a592e388cd9db5b5c456a7ae0922b9a054c8d6ba

**Comment by nassissme** [comment]
Thank you for considering my problem.
